addons/sale/wizard/sale_make_invoice_advance.py

```diff
@@ -221,7 +221,7 @@ def _create_invoices(self, sale_orders):
 
             invoice = self.env['account.move'].sudo().create(
                 self._prepare_invoice_values(order, down_payment_lines)
-            ).with_user(self.env.uid)  # Unsudo the invoice after creation
+            ).with_user(self.env.uid).sudo(self.env.su)  # Unsudo the invoice after creation if not already sudoed
```

Comment:
```suggestion
            ).sudo(self.env.su)  # Unsudo the invoice after creation if not already sudoed
```
User update is unnecessary now :D
Also, this `.sudo(self.env.su)` could be moved after the 'if self.advance_payment_method == 'fixed':` block (we want to create the invoice in sudo, but also update it in sudo)